libsolidity/parsing/Parser.cpp

```diff
@@ -111,7 +111,17 @@ ASTPointer<SourceUnit> Parser::parse(shared_ptr<Scanner> const& _scanner)
 				nodes.push_back(parseFunctionDefinition(true));
 				break;
 			default:
-				fatalParserError(7858_error, "Expected pragma, import directive or contract/interface/library/struct/enum/function definition.");
+				// Constant variable.
+				if (variableDeclarationStart())
+				{
+					VarDeclParserOptions options;
+					options.isStateVariable = true;
```

Comment:
Why are file level constants considered state variables?

Reply:
This is just for the parser. This allows to parse the 'constant' keyword. I could also heavily change that method, but I thought it better to add extensive checks in the analysis phase later.

Should I rename the option field to `isStateOrFileLevelVariable`?

Reply:
It's fine, just wanted to check.

Reply:
Maybe renaming it wouldn't be the worst of ideas? I guess it could potentially spare us quite some headaches in the future...

Reply:
Also not sure if a ``isFileLevelVariable`` field in ``VarDeclParserOptions`` would have been that bad... then ``parseVariableDeclaration`` could just reject docstrings, have a better error message for the function type thing, and reject visibility and override...
But yeah - maybe doing all of that in analysis has advantages - one could argue either way I guess and both is fine with me.